unit_tests/noddy/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+set(TEST_NODDY_SRC_FILES Holder.cpp)
+
+source_group(test\\noddy FILES ${TEST_SRC_FILES})
+
+FIND_PACKAGE(Threads REQUIRED)
+
+# from list of files we'll create tests test_name.cpp -> test_name
+foreach(_test_file ${TEST_NODDY_SRC_FILES})
+    get_filename_component(_test_name ${_test_file} NAME_WE)
+    add_executable(${_test_name} ${_test_file})
+    target_link_libraries(${_test_name} gtest_main ${CMAKE_THREAD_LIBS_INIT})
+    add_test(${_test_name} ${_test_name})
+endforeach()
```

Comment:
Instead of `add_test` can we use `gtest_discover_tests`?  This improves the potentials for parallelism?

Reply:
Done, but removed the third party directory so that all the Google Test stuff is pulled into the correct directory. So have done a force push rather than add and remove.